packages/xo-server/src/api/pool.mjs

```diff
@@ -163,12 +165,42 @@ installPatches.description = 'Install patches on hosts'
 // -------------------------------------------------------------------
 
 export const rollingUpdate = deferrable(async function ($defer, { pool }) {
-  if ((await this.getOptionalPlugin('load-balancer'))?.loaded) {
-    await this.unloadPlugin('load-balancer')
-    $defer(() => this.loadPlugin('load-balancer'))
-  }
+  const poolId = pool.id
+  const [schedules, jobs] = await Promise.all([this.getAllSchedules(), this.getAllJobs('backup')])
+
+  const jobsOfthePool = []
+  jobs.forEach(({ id: jobId, vms }) => {
+    if (vms.id !== undefined) {
+      for (const vmId of extractIdsFromSimplePattern(vms)) {
+        // try/catch to avoid `no such object`
+        try {
+          if (this.getObject(vmId).$poolId === poolId) {
+            jobsOfthePool.push(jobId)
+            break
+          }
+        } catch {}
+      }
+    } else {
+      // SMARTMODE
```

Comment:
I think we should add the comment explaining it also here : 
```suggestion
// smartmode
// For the smartmode we take a simplified approach :
// if the smartmode is explicitly 'resident' or 'not resident' on pools : we check if it concern this pool 
// if not, the job  may concern this pool and we show the warning without looking through all the impacted VM
```

Reply:
```suggestion
      // smartmode
	// For the smartmode we take a simplified approach :
	// if the smartmode is explicitly 'resident' or 'not resident' on pools : we check if it concern this pool 
	// if not, the job  may concern this pool so we add him to the `jobsOfThePool`
```